spec/javascripts/govuk_publishing_components/rum/http-protocol-measurement-spec.js

```diff
@@ -0,0 +1,26 @@
+/* eslint-env jasmine */
+/* global LUX */
+
+// This test will only work if the browser running it supports the HTML5
+// Performance API.
+if (typeof window.performance !== 'undefined') {
+  describe('The LUX RUM function should report the the HTTP protocol', function () {
```

Comment:
I'm not sure I understand how this test works, although my knowledge of this whole area is limited. I guess I'm used to tests that create a condition then compare the outcome with an expected set of data. But this test doesn't create a condition - it just expects things. Is there any comment you could add to explain this a little?

Reply:
I've added some more in depth detail about this that I hope makes sense. 

It's a difficult and different thing to test, as the script runs automatically and shouldn't require initialisation because it should run as soon as possible to get the most accurate measurement. Unfortunately I can't think of a better way of testing this.